File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

int solution(int number, int limit, int power) {
    int answer = 0;
    
    vector<int> attack;
    for(int i=1; i<=number; i++){
        // 약수 구하기
        int div = 1; // 자기 자신
        for(int j=1; j<=i/2; j++){
            if(i%j == 0)
                div++;  
        }
        
        div = div>limit ? power : div;
        attack.push_back(div);
    }
    
    for(int i=0; i<attack.size(); i++)
        answer += attack[i];
    
    return answer;
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {
    var answer = 0;
    
    for(let i=1; i<=number; i++){
        let cnt = 0;

        // 약수 개수
        for(let j = 1; j <= Math.sqrt(i); j++){
            if(i % j === 0){
                cnt += 2;
                
                if(i/j === j)
                    cnt -= 1;  
            }
            
            if(cnt > limit){
                cnt = power;
                break;
            }
        }
        
        answer += cnt;
    }
    
    return answer;
}